farm/data_handler/processor.py

```diff
@@ -608,66 +608,98 @@ def file_to_dicts(self, file: str) -> [dict]:
 
         return dicts
 
-    def _dict_to_samples(self, dictionary: dict, **kwargs) -> [Sample]:
-        # this tokenization also stores offsets and a start_of_word mask
-        text = dictionary["text"]
-        tokenized = tokenize_with_metadata(text, self.tokenizer)
-        if len(tokenized["tokens"]) == 0:
-            logger.warning(f"The following text could not be tokenized, likely because it contains a character that the tokenizer does not recognize: {text}")
-            return []
-        # truncate tokens, offsets and start_of_word to max_seq_len that can be handled by the model
-        for seq_name in tokenized.keys():
-            tokenized[seq_name], _, _ = truncate_sequences(seq_a=tokenized[seq_name], seq_b=None, tokenizer=self.tokenizer,
-                                                max_seq_len=self.max_seq_len)
-        return [Sample(id=None, clear_text=dictionary, tokenized=tokenized)]
-
-    def _sample_to_features(self, sample) -> dict:
-        features = sample_to_features_text(
-            sample=sample,
-            tasks=self.tasks,
-            max_seq_len=self.max_seq_len,
-            tokenizer=self.tokenizer,
-        )
-        return features
-
-    def _dict_to_samples_and_features(self, dictionary: dict, **kwargs) -> [Sample]:
-        """This method is used so that we need to tokenize only once when using a fast tokenizer."""
-        text = dictionary["text"]
-        inputs = self.tokenizer.encode_plus(
-            text=text,
-            max_length=self.max_seq_len,
-            truncation=True,
-            add_special_tokens=True,
+    def dataset_from_dicts(self, dicts, indices=None, return_baskets=False, inference=False, debug=False):
```

Comment:
We should document the reasoning / behaviour of `inference` flag here.

Reply:
After a short discussion with @Timoeller, we decided not to change the signature of `datset_from_dicts( )` yet since changing it for one task will require it to be changed for all. `return_baskets` is a good enough proxy for `inference` for now. We will change that naming at a later stage